be/src/formats/parquet/parquet_file_writer.cpp

```diff
@@ -48,7 +49,13 @@ std::future<Status> ParquetFileWriter::write(ChunkPtr chunk) {
     if (auto status = _rowgroup_writer->write(chunk.get()); !status.ok()) {
         return make_ready_future(std::move(status));
     }
-    if (_rowgroup_writer->estimated_buffered_bytes() >= _writer_options->rowgroup_size) {
+    double mem_usage = 0.0;
```

Comment:
It's better to implement this in `ConnectorSinkOperator` so all file formats can benefit from this